i3_agenda/__init__.py

```diff
@@ -2,359 +2,76 @@
-    type=int,
-    default=-1,
-    help="minutes to show events after they start before showing the next event. If not specified, the current event will be shown until it ends",
-)
-parser.add_argument(
-    "--date-format",
-    type=str,
-    default="%d/%m",
-    help="the date format like %%d/%%m/%%y. Default is %%d/%%m",
-)
-parser.add_argument(
-    "--limchar",
-    "-l",
-    type=int,
-    default=-1,
-    help="the max characters that the displayed event can contain",
-)
+from typing import List
 
+from event import Event, EventEncoder
+from config import parser, button
+from event_utils import get_closest
+from API import getEvents
+from cache_utils import load_cache, save_cache
 
-class Event:
-    def __init__(
-        self,
-        summary: str,
-        is_allday: bool,
-        unix_time: float,
-        end_time: float,
-        location: str,
-    ):
-        self.is_allday = is_allday
-        self.summary = summary
-        self.unix_time = unix_time
-        self.end_time = end_time
-        self.location = location
-
-
-class EventEncoder(json.JSONEncoder):
-    def default(self, o):  # pylint: disable=E0202
-        if isinstance(o, Event):
-            return o.__dict__
-        else:
-            return json.JSONEncoder.default(self, o)
+DEFAULT_CAL_WEBPAGE = "https://calendar.google.com/calendar/r/day"
 
 
-def main():
-    args = parser.parse_args()
+def button_action(button_code: str, closest: Event):
+    if button_code != "":
+        if button_code == "1":
+            print("Opening calendar page...")
+            subprocess.Popen(["xdg-open", DEFAULT_CAL_WEBPAGE])
+        elif button_code == "3":
+            print("Opening location link...")
+            subprocess.Popen(["xdg-open", closest.location])
 
-    allowed_calendars_ids = args.ids
 
+def get_events(args) -> List[Event]:
     events = None
     if not args.update:
         events = load_cache(args.cachettl)
-    if events == None:
-        service = connect(args.credentials)
-        events = getEvents(service, allowed_calendars_ids, args.maxres, args.today)
+    if events is None:
+        events = getEvents(args.credentials, args.ids, args.maxres, args.today)
         save_cache(events)
+    return events
 
-    closest = get_closest(events, args.hide_event_after)
-    if closest is None:
-        print(args.no_event_text)
-        return
 
-    if button != "":
-        if button == "1":
-            print("Opening calendar page...")
-            subprocess.Popen(["xdg-open", DEFAULT_CAL_WEBPAGE])
-        elif button == "3":
-            print("Opening location link...")
-            subprocess.Popen(["xdg-open", closest.location])
-
-    event = datetime.datetime.fromtimestamp(closest.unix_time)
-    today = datetime.datetime.today()
-    tomorrow = today + datetime.timedelta(days=1)
-    next_week = today + datetime.timedelta(days=7)
-    urgent = today + datetime.timedelta(minutes=5)
+def get_event_string(closest: Event, args) -> str:
```

Comment:
no reason this function should be in the __init__ file, cold be separated into an "EventFormatter" object or even as a method of the "Event" class, in both cases instead of receiving `args` it could receive only the relevant information (chars limit, date format)

Reply:
Done